### DISCUSSION TITLE
2 simple feature requests - I will pay for it

### DISCUSSION THREAD

**Discussion by cribaudo** [BLeeEZ/amperfy#258] [Ideas]
Hi, our non-profit organization uses playlists for music at our events. Apple has been messing around with iTunes so much that we can no longer use it and have migrated to Navidrome & Amperfy. 

I would be happy to make a contribution to this project for the following 2 simple enhancements:

1. Add a setting to Amperfy Settings to optionally HIDE the Shuffle button. We don't ever use it or want the possibility of accidentally hitting it.
2. Could also be a setting in Amperfy Settings. Do NOT automatically play a track when using the next track button to skip to the next track in a playlist. We typically pull the play list up on the device and then sometimes have to move back and forth between tracks to queue the next one up for when it is needed. It would be perfect for us if they track one skipped to did not automatically play.

These two features will make the app ABSOLUTELY perfect for us.

**Comment by BLeeEZ** [comment]
Hi, I implemented both of your requests:
1: implemented in c41a012: 
![Simulator Screenshot - iPhone 15 Pro Max - 2024-05-20 at 21 20 31](https://github.com/BLeeEZ/amperfy/assets/17556018/c166cb94-c78a-46f9-9678-76de6beffdd2)
2: implemented in 7023a61:
![Simulator Screenshot - iPhone 15 Pro Max - 2024-05-20 at 21 43 20](https://github.com/BLeeEZ/amperfy/assets/17556018/ca38d0c1-8321-496a-8b9f-5d1050930ea7)

**Comment by cribaudo** [comment]
Amazing what version will it appear in in the app store? I ll make a $500 donation to the project. Thanks. iPhemerisOn May 20, 2024, at 10:21 PM, Maximilian Bauer ***@***.***> wrote:﻿
Hi, I implemented both of your requests:
1: implemented in c41a012:
Simulator.Screenshot.-.iPhone.15.Pro.Max.-.2024-05-20.at.21.20.31.png (view on web)
2: implemented in 7023a61:
Simulator.Screenshot.-.iPhone.15.Pro.Max.-.2024-05-20.at.21.43.20.png (view on web)

**Comment by BLeeEZ** [comment]
These features can be tested in the newest beta v1.2.1 (2): https://github.com/BLeeEZ/amperfy/issues/25

**Comment by BLeeEZ** [comment]
Did you have time test the feature? Is it behaving as you expected?

**Comment by cribaudo** [comment]
I didn't have time to test but finally figured out how to send you $100 using Paypal. Thanks. Ill check it out when it is production.

On Jun 15, 2024, at 5:11 AM, Maximilian Bauer ***@***.***> wrote:


Did you have time test the feature? Is it behaving as you expected?

**Comment by erikmakela** [comment]
Thank you for the project, I recently setup my own server. Both features are working as intended as of current.

To add, I'd recommend disabling the random playlist interaction within the Library section (such as when going through songs or directories). As misclicks could still happen when trying to find a new song with say a different album.

**Comment by BLeeEZ** [comment]
Thank you. I am not completely getting your improvement idea. Do you mean deactivating the „Play“ and „Shuffle“ buttons on the top in those library categories?

**Comment by erikmakela** [comment]
@BLeeEZ Yes, Exactly.
